// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package teeapps.params;

message LrHyperParams {
  // Maximum number of iterations taken for the solvers to converge
  //  range: [1, 1e4]
  //  default: 10
  int64 max_iter = 1;
  // Regression type
  //  allowed: {"linear", "logistic"}
  //  default: "logistic"
  string regression_type = 2;
  // L2 regularization term
  //  range: [0, 1e4)
  //  default: 1.0
  float l2_norm = 3;
  // Tolerance for stopping criteria
  //  range: (0, 1)
  //  default: 1e-4
  float tol = 4;
  // The penalty(aka regularization term) to be used
  //  allowed: {"l1", "l2", "elasticnet", "None"}
  //  default: "l2"
  string penalty = 5;
}
